naivedb_test.go

```diff
@@ -0,0 +1,44 @@
+package naivedb
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestGetBeforeSet(t *testing.T) {
+	buf := new(bytes.Buffer)
+	db := NaiveDB{buf}
+
+	key := "foo"
+	stored_value, err := db.get(key)
+
+	if err != nil {
+		t.Fatalf(`%q`, err)
+	}
+
+	if stored_value != "" || err != nil {
+		t.Fatalf(`Expected set to return %q but got %q, %v`, "", stored_value, err)
+	}
+}
+
+func TestSetThenGet(t *testing.T) {
+	buf := new(bytes.Buffer)
+	db := NaiveDB{buf}
+
+	key := "foo"
+	value := "bar"
+
+	err := db.set(key, value)
+
+	// TODO why does the sample code use a string literal?
+	// https://golang.org/doc/tutorial/add-a-test
+	if err != nil {
+		t.Fatalf(`%q`, err)
```

Comment:
Still not sure why they use the string literal `.

Reply:
I think in the sample they use backticks to surround the string because the string contains double-quotes, and this way they don't have to escape them.

Reply:
Aha! Thank you.